cmake/compiler_settings.cmake

```diff
@@ -27,7 +27,7 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 # Set default build type and sanitize.
 # TODO: See #756: Fix this since debug is the default.
 if (NOT CMAKE_BUILD_TYPE)
-  set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "Build type" FORCE)
```

Comment:
This change is not necessary given the reason in the PR description:
* This affects the default build option for contributors building the OE SDK directly, and in those cases, building Debug by default is more generally useful for the developers. The build system already builds RelWithDebInfo for OE SDK releases.
* The `CMAKE_BUILD_TYPE` only controls the compiler settings (optimization levels and inclusion of symbols), which is distinct from the enclave's `ALLOW_DEBUG` property set by `OE_SET_ENCLAVE_SGX` or by oesign. The latter flag is the security concern and is what allows the enclave to be created with the `OE_ENCLAVE_FLAG_DEBUG` flag that allows debuggers to attach to and introspect the enclave.
    * It is a specific use case to allow a Release compiled enclave binary set with the `ALLOW_DEBUG` flag for testing with a test signature, then re-sign the enclave with a different production key and the `ALLOW_DEBUG` flag disabled without having to recompile the enclave.

Reply:
Thanks! For the host side, most open source releases always turn off debug mode, since debug mode may have detailed log output. Since enclave side has more security check, this is not a security issue. I will withdraw this PR.